pkg/route/routeapihelpers/routeapihelpers.go

```diff
@@ -42,3 +44,30 @@ func IngressURI(route *routev1.Route, host string) (*url.URL, *routev1.RouteIngr
 	}
 	return nil, nil, fmt.Errorf("no ingress for host %s in route %s in namespace %s", host, route.ObjectMeta.Name, route.ObjectMeta.Namespace)
 }
+
+// ValidateHost checks that a route's host name satisfies DNS requirements, with
+// the assumption that the caller has already checked for an empty host name.
+// Unless the allowNonCompliant annotation is set to true, host name must have
+// at least two labels, with each label no more than 63 characters from the set of
+// alphanumeric characters, '-' or '.', and must start and end with an alphanumeric
+// character. A trailing dot is allowed. The total host name length must be no more
```

Comment:
Sorry, I missed this before; the actual rule is that it must start with an alphabetic character and end with an alphanumeric character.

Reply:
I don't think you missed it, in https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go#L177 we allow it to start with alphanumeric, not alphabetic.

Reply:
Ah, you're right.  I was looking at 1035, which is more restrictive than 1123.